dpnp/linalg/dpnp_iface_linalg.py

```diff
@@ -502,6 +509,59 @@ def qr(x1, mode="reduced"):
     return call_origin(numpy.linalg.qr, x1, mode)
 
 
+def solve(a, b):
+    """
+    Solve a linear matrix equation, or system of linear scalar equations.
+
+    For full documentation refer to :obj:`numpy.linalg.solve`.
+
+    Returns
+    -------
+    out : {(…, M,), (…, M, K)} dpnp.ndarray
+        Solution to the system ax = b. Returned shape is identical to b.
+
+    Limitations
+    -----------
+    Parameters `a` and `b` are supported as either :class:`dpnp.ndarray`
+    or :class:`dpctl.tensor.usm_ndarray`.
+    Input array data types are limited by supported DPNP :ref:`Data types`.
+
+    See Also
+    --------
+    :obj:`dpnp.dot` : Returns the dot product of two arrays.
+
+    Examples
+    --------
+    >>> import dpnp as dp
+    >>> a = dp.array([[1, 2], [3, 5]])
+    >>> b = dp.array([1, 2])
+    >>> x = dp.linalg.solve(a, b)
+    >>> x
+    array([-1.,  1.])
+
+    Check that the solution is correct:
+
+    >>> dp.allclose(dp.dot(a, x), b)
+    array([ True])
+
+    """
+
+    _assert_supported_array_type(a, b)
+    _assert_stacked_2d(a)
+    _assert_stacked_square(a)
```

Comment:
I would suggest either to move that inside `dpnp/linalg/dpnp_utils_linalg.py` or to rename the functions to make them external and pointing to linalg specific things

Reply:
I renamed these functions and made them external to `dpnp/linalg/dpnp_utils_linalg.py`
Thanks